foundation/foundation-upgrade/src/test/java/com/boozallen/aissemble/upgrade/migration/utils/YamlUtilsSteps.java

```diff
@@ -0,0 +1,96 @@
+package com.boozallen.aissemble.upgrade.migration.utils;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.util.YamlUtils;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.When;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class YamlUtilsSteps {
+
+    private Path testFile;
+    private YamlUtils.YamlObject yaml;
+
+    @Given("a YAML file:")
+    public void aYamlFile(String contents) throws IOException {
+        testFile = Files.createTempFile(Path.of("target"), "yaml-util-test", "");
+        Files.writeString(testFile, contents);
+    }
+
+    @When("the file is loaded")
+    public void theFileIsLoaded() throws IOException {
+        yaml = YamlUtils.loadYaml(testFile.toFile());
+    }
+
+    @Then("the string value of the property {string} is {string}")
+    public void theStringValueOfThePropertyIs(String name, String value) {
+        String type = yaml.get(name).getClass().getSimpleName();
+        assertTrue("Property was not loaded as string: " + name + " (was " + type + ")", yaml.hasString(name));
+        assertEquals("Property value does not match for: " + name, value, yaml.getString(name));
+    }
+
+    @Then("the decimal value of the property {string} is {double}")
+    public void theDecimalValueOfThePropertyIs(String name, double value) {
+        String type = yaml.get(name).getClass().getSimpleName();
+        assertTrue("Property was not loaded as double: " + name + " (was " + type + ")", yaml.hasDouble(name));
+        assertEquals("Property value does not match for: " + name, value, yaml.getDouble(name), 0.01);
```

Comment:
Q: `0.01`?

Reply:
It's a `delta` param that they added to floating point assertions. Avoids the tricky floating point rounding issues where you get `0.000000000001` or some other minuscule difference just from how FP is represented. So this says treat the numbers as equal if they are within `0.01` of each other.